Assets/Player/PlayerHud.gd

```diff
@@ -15,9 +17,25 @@ var _debug_messages = [
 	[4, "This is a very long text. That much, that it easily takes up to 3 lines. Believe it or not."]
 ]
 
+var old_selected_units = null
+
+func _ready():
+	EventBus.connect("selected", self, "_on_selected")
+
 func _process(_delta):
 	pass
 
+	
+func _on_selected():
+	var selected_units = get_tree().get_nodes_in_group("selected_units")
+	if selected_units.size() == 0:
+		# should in the future hide all unit-related HUD-Elements
+		tab_widget._set_detail_visibility(false)
+	elif selected_units.size() >= 1: 
+		tab_widget._change_on_select(selected_units)
+	
+		
+
```

Comment:
1. The style convention we are using for this project is to have 1 line between functions. Here you have 3.

2. Private functions (ones that start with an underscore that aren't *lifecycle* functions like _ready or _process) should go after public ones.